### ISSUE TITLE
PHP 8.4 future release and support to property hooks

### ISSUE THREAD

**Issue by peregrinodeti** [eclipse-pdt/pdt#279]
**Is your feature request related to a problem? Please describe.**
PHP 8.4 it's schedule to be release at november 2024. But your alpha's will be release in July. It's any planning to support property hooks at Eclipse 2024-09 release or to build nights to current version? 

**Is this feature require new UI elements or modify existing?**
No.

**Describe the solution you'd like**
One of the most anticipated features of PHP 8.4 is property hooks. This feature allows developers to define custom "get" and "set" behaviors of class properties. Inspired by languages like Kotlin and Swift, property hooks simplify the creation of accessors and mutators, reducing boilerplate code and making the codebase cleaner and more maintainable.

Example: 

```php
class Foo
{
    private int $runs = 0;
 
    public function getRuns(): int { return $this->runs; }
 
    public function setRuns(int $runs): void
    {
      if ($runs <= 0) throw new Exception();
      $this->runs = $runs;
    }
}
 
$f = new Foo();
 
$f->setRuns($f->getRuns() + 1);
``` 

**Describe alternatives you've considered**
Implement support this language feature, in next or actual Eclipse version. If need help to test or something @zulus please let me know.

**Additional context**
Reference about this feature in this RFC: https://wiki.php.net/rfc/property-hooks

**Comment by peregrinodeti** [comment]
Well, PHP 8.4 Alpha 1 was released last week. I tried to use it in Eclipse PDT 2024-06, but when you go to the Eclipse IDE menu: Window -> Preferences -> Installed PHPs and try to point to the new PHP 8.4, the IDE throws an exception (Selected location doesn't refer to a valid PHP executable). The problem is not the location of the file, but the new version of PHP. I test on Windows 11. PHP 8.3.9 works fine when using the same directory name; in my case, c:\php. In both cases, the PHP executable stays in the same directory, but 8.4 doesn't work at the IDE, the IDE does not recognize the new version of the PHP executable.

If not difficult, I suggest a new IDE UI option to generate property hooks, like get and set. When you click on the source code and a sub-menu appears, go to the source option and put the new option: **"Property Hooks Get and Set"**, for example. 

@zulus If you need any help to testing this new feature, let me know. And thanks for your great work at IDE.

**Comment by peregrinodeti** [comment]
Well, Update about the hooks. I use Eclipse IDE 2025-06, PHP 8.4.8 on Windows 11. I try to use property hooks, but the last version of the Eclipse PDT don't recognize this feature. Follow the code and errors:

```php
<?php
class Category
{
    public ?int $id = null {
        get => $this->id;
        set => $this->id = $value;
    }
    
    public string $description = '' {
        get => $this->description;
        set => $this->description = $value;
    }
}

$test = new Category();
$test->id = 1000;
$test->description = "Test";

var_dump($test);
```

**Internal in Eclipse PDT, I have these errors:**

Description	Resource	Path	Location	Type
syntax error, unexpected '=>'	Category.php	/hook840	line 5	PHP Problem

Description	Resource	Path	Location	Type
syntax error, unexpected '=>'	Category.php	/hook840	line 10	PHP Problem

I test the code, the code works fine, but the IDE don't recognize this feature. I test the code on other IDEs and then recognize the feature. 

@zulus Again, if you need any help to testing this new feature, let me know. And thanks for your great work at IDE.

**Comment by peregrinodeti** [comment]
> I'd very much like to see this feature as well, I'm going to dramatically need it within the few weeks/months to come.

Yes me too. I very concerned that the project will be frozen again, I post this comment a 1 year ago. And by the end of 2025 we will have PHP 8.5 and new features.

**Comment by pounard** [comment]
We need this! @zulus any news? You haven't been active for a while, if you need help please ask.

**Comment by zulus** [comment]
I finally grab some free time to play with this. I also need this. At first I have to fix some parser issues ;)
